shpc/main/registry/remote.py

```diff
@@ -231,10 +205,48 @@ def iter_registry(self, filter_string=None):
             # Assemble a faux config with tags so we don't hit remote
             yield RemoteResult(uri, entry, load=False, config=entry["config"])
 
+    def get_container_url(self, module_name):
+        raise NotImplementedError
+
+    def get_raw_container_url(self, module_name):
+        raise NotImplementedError
+
 
 class GitHub(VersionControl):
-    provider_name = "github"
+    @classmethod
+    def matches(cls, source):
+        return urllib.parse.urlparse(source).hostname == "github.com"
+
+    @property
+    def library_url(self):
+        url_path = self.parsed_url.path.lstrip("/")
+        if url_path.endswith(".git"):
+            url_path = url_path[:-4]
+        owner, repo = url_path.split("/", 1)
+        return f"https://{owner}.github.io/{repo}/library.json"
```

Comment:
These library_url properies are almost duplicates of one another, minus the repository name "github" or "gitlab." Should we possibly have a remote_name (or similar) attribute that goes into a shared function?

Reply:
I didn't do exactly that, but I could shorten both implementations of `library_url` quite a bit.
I want to avoid the base-class knowing too much about its sub-classes and how GitHub/GitLab work